### DISCUSSION TITLE
Interface with heater documentation

### DISCUSSION THREAD

**Discussion by Reeter35** [openv/openv#513] [General]
Hi,
Not being a german speaker, I am a bit lost in the wiki: do you have, somewhere, a documentation about the optolink interface (how to communicate with the heater using an optloink/USB interface, especially on some models such as Vitoligno) ?

Thanks,

**Comment by bertmelis** [comment]
The possibility of using an optolink depends on the control panel. It needs to have the "V with optical port". If that is available, it's a matter of finding the datapoints you can read or write.

what kind of information are you looking for? Hardware, software?

**Reply by Reeter35** [reply]
I think this is an optolink connector: actually I have a vitoconnect connected to my heater (through what seems to be a Opto to USB cable), and I would like to get rid of it because the Vicare app does not display nor allow me to manage all the data I would like.

So I had a look to openv, and saw vitalk and vitocontrold (but they seems to communicate with vitodens): I was wondering if you had a brief description of the communication protocol with the heater, so that I could try to make it work with a vitoligno, or so that I try to develop my own script to retrive the data ana analyze them later.

**Reply by bertmelis** [reply]
Ok.

Thing is that Viessmann doesn't document the available datapoints/memory locations. Each piece of information has it's address and length and format. A bit like modbus. However, unless you have a list of these datapoints, you have to search for it yourself and this is like reverse engineering. No success guaranteed.

There is a possibility however. I don't know if it works for newer systems. Viessmann has professional installers software with a  day trial. You can try this, update the data files (these are large xml-files) and extract the data. I don't know if it's still available for download though.
